apps/fabric-website/src/SiteDefinition/SiteDefinition.pages/Controls/web.tsx

```diff
@@ -1,580 +1,348 @@
+  Surfaces: {
+    Callout: {},
+    Dialog: {},
+    Modal: {},
+    Panel: {},
+    ScrollablePane: {},
+    Tooltip: {}
   },
-  {
-    title: 'Surfaces',
-    url: '#/controls/web/callout',
-    isCategory: true,
-    pages: [
-      {
-        title: 'Callout',
-        url: '#/controls/web/callout',
-        component: () => <LoadingComponent title="Callout" />,
-        getComponent: cb => require.ensure([], require => cb(require<any>('../../../pages/Controls/CalloutPage/CalloutPage').CalloutPage))
-      },
-      {
-        title: 'Dialog',
-        url: '#/controls/web/dialog',
-        component: () => <LoadingComponent title="Dialog" />,
-        getComponent: cb => require.ensure([], require => cb(require<any>('../../../pages/Controls/DialogPage/DialogPage').DialogPage))
-      },
-      {
-        title: 'Modal',
-        url: '#/controls/web/modal',
-        component: () => <LoadingComponent title="Modal" />,
-        getComponent: cb => require.ensure([], require => cb(require<any>('../../../pages/Controls/ModalPage/ModalPage').ModalPage))
-      },
-      // {
-      //   title: 'Panel',
-      //   url: '#/controls/web/panel',
-      //   component: () => <LoadingComponent title="Panel" />,
-      //   getComponent: cb => require.ensure([], require => cb(require<any>('../../../pages/Controls/PanelPage/PanelPage').PanelPage))
-      // },
-      {
-        title: 'Tooltip',
-        url: '#/controls/web/tooltip',
-        component: () => <LoadingComponent title="Tooltip" />,
-        getComponent: cb => require.ensure([], require => cb(require<any>('../../../pages/Controls/TooltipPage/TooltipPage').TooltipPage))
-      },
-      {
-        title: 'ScrollablePane',
-        url: '#/controls/web/scrollablepane',
-        component: () => <LoadingComponent title="ScrollablePane" />,
-        getComponent: cb =>
-          require.ensure([], require =>
-            cb(require<any>('../../../pages/Controls/ScrollablePanePage/ScrollablePanePage').ScrollablePanePage)
-          )
+  Utilities: {
+    Announced: {
+      subPages: {
+        QuickActions: { title: 'Quick Actions' },
+        SearchResults: { title: 'Search Results' },
+        LazyLoading: { title: 'Lazy Loading' },
+        BulkOperations: { title: 'Bulk Operations' }
       }
-    ]
+    },
+    FocusTrapZone: {},
+    FocusZone: {},
+    Icon: {},
+    Image: {},
+    Layer: {},
+    MarqueeSelection: {},
+    Overlay: {},
+    ResizeGroup: {},
+    Selection: {},
+    Separator: {},
+    Stack: {},
+    Text: {},
+    Themes: {}
   },
-  {
-    title: 'Fluent Theme',
-    url: '#/controls/web/fluent-theme',
-    component: () => <LoadingComponent title="Fluent Theme" />,
-    getComponent: cb =>
-      require.ensure([], require => cb(require<any>('../../../pages/Controls/FluentThemePage/FluentThemePage').FluentThemePage))
+  'Fluent Theme': {
+    FluentTheme: { title: 'Fluent Theme', url: 'fluent-theme' }
+  },
+  Other: {}
+};
+
+function generateCategories() {
```

Comment:
If possible, I'd love to generalize this more in the future and use it to generate all the parts of the config.